api/apps/api/src/modules/scenarios/scenarios.controller.ts

```diff
@@ -408,21 +410,43 @@ export class ScenariosController {
     await ensureShapefileHasRequiredFiles(file);
 
     const result = await this.service.processCostSurfaceShapefile(
-      scenarioId,
+      req.params.scenarioId,
       req.user.id,
       file,
     );
 
     if (isLeft(result)) {
       throw mapAclDomainToHttpError(result.left, {
-        scenarioId,
+        scenarioId: req.params.scenarioId,
         userId: req.user.id,
         resourceType: scenarioResource.name.plural,
       });
     }
     return AsyncJobDto.forScenario().asJsonApiMetadata();
   }
 
+  @ApiOperation({
+    deprecated: true,
+    description:
+      'To be removed soon to GET /projects/:projectId/cost-surface/:costSurfaceId/cost-range',
+  })
+  @Get(`:id/cost-surface`)
+  @ApiOkResponse({ type: CostRangeDto })
+  async getCostRange(
+    @Param('id') scenarioId: string,
+    @Req() req: RequestWithAuthenticatedUser,
+  ): Promise<CostRangeDto> {
+    const result = await this.service.getCostRange(scenarioId, req.user.id);
+    if (isLeft(result)) {
+      throw mapAclDomainToHttpError(result.left, {
+        scenarioId,
+        userId: req.user.id,
+        resourceType: scenarioResource.name.plural,
+      });
+    }
+    return plainToClass<CostRangeDto, CostRangeDto>(CostRangeDto, result.right);
+  }
+
```

Comment:
This is totally my fault, as it was removed in f1e3d8de931e6b3f4b2410ff8ef3731736ab7c34 as part of:
[MARXN-23-92](https://vizzuality.atlassian.net/browse/MRXN23-92)
Where it was required to remove all cost-surface related operations to the new controller. I assumed it was a safe operation. 

Thanks!

PS: It's double-weird that in the previous endpoint the `@Param()` decorator seems to not be working, and we expect here to do it. Let's discuss it later